lib/promscrape/scrapework.go

```diff
@@ -107,6 +107,8 @@ type ScrapeWork struct {
 	// ProxyURL HTTP proxy url
 	ProxyURL *proxy.URL
 
+	ProxyForceHTTPConnect bool
```

Comment:
Is this variable needed here or only one defined in `ProxyClientConfig` is needed?

Reply:
So `ScrapeConfig` has a `ProxyClientConfig` which is a natural place to put this. However that config gets translated to a generic struct `promauth.Config` on both `ScrapeWorkConfig` and `ScrapeWork`.

The `promauth.Config` is used for both destination endpoints and proxying, and is just TLS info. Adding it there would leave a dangling appendage on the regular `AuthConfig` field.

That said, I'm open to doing whatever the VM team recommends :)

Reply:
Thank you for the details, that makes sense to me.

Could you please add a comment for this parameter to match the code style?

Reply:
done!